src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisTest.java

```diff
@@ -1086,4 +1086,133 @@ void testNullBusInjection() {
 
         assertEquals(92.0836, result.getBranchCurrent1FunctionReferenceValue("NHV1_NHV2_1"), LoadFlowAssert.DELTA_I);
     }
+
+    @Test
+    void testThreeWindingsTransformerAsFunction() {
+        SensitivityAnalysisParameters sensiParameters = createParameters(false, "b1_vl_0", true);
+        sensiParameters.getLoadFlowParameters().setBalanceType(LoadFlowParameters.BalanceType.PROPORTIONAL_TO_GENERATION_P_MAX);
+        Network network = VoltageControlNetworkFactory.createNetworkWithT3wt();
+        SensitivityFactor factorActivePower1Twt = new SensitivityFactor(SensitivityFunctionType.BRANCH_ACTIVE_POWER_1, "T3wT",
+                SensitivityVariableType.INJECTION_ACTIVE_POWER, "LOAD_4", false, ContingencyContext.all());
+        SensitivityFactor factorActivePower2Twt = new SensitivityFactor(SensitivityFunctionType.BRANCH_ACTIVE_POWER_2, "T3wT",
+                SensitivityVariableType.INJECTION_ACTIVE_POWER, "LOAD_4", false, ContingencyContext.all());
+        SensitivityFactor factorActivePower3Twt = new SensitivityFactor(SensitivityFunctionType.BRANCH_ACTIVE_POWER_3, "T3wT",
+                SensitivityVariableType.INJECTION_ACTIVE_POWER, "LOAD_4", false, ContingencyContext.all());
+
+        SensitivityFactor factorCurrent1 = new SensitivityFactor(SensitivityFunctionType.BRANCH_CURRENT_1, "T3wT",
+                SensitivityVariableType.INJECTION_ACTIVE_POWER, "LOAD_4", false, ContingencyContext.all());
+        SensitivityFactor factorCurrent2 = new SensitivityFactor(SensitivityFunctionType.BRANCH_CURRENT_2, "T3wT",
+                SensitivityVariableType.INJECTION_ACTIVE_POWER, "LOAD_4", false, ContingencyContext.all());
+        SensitivityFactor factorCurrent3 = new SensitivityFactor(SensitivityFunctionType.BRANCH_CURRENT_3, "T3wT",
+                SensitivityVariableType.INJECTION_ACTIVE_POWER, "LOAD_4", false, ContingencyContext.all());
+
+        List<SensitivityFactor> factors = List.of(factorActivePower1Twt, factorActivePower2Twt, factorActivePower3Twt,
+                factorCurrent1, factorCurrent2, factorCurrent3);
+        SensitivityAnalysisResult result = sensiRunner.run(network, factors, Collections.emptyList(), Collections.emptyList(), sensiParameters);
+        assertEquals(6, result.getValues().size());
+
+        //Function active power
+        assertEquals(-1.001, result.getValues().get(0).getValue(), LoadFlowAssert.DELTA_POWER);
```

Comment:
Can you write the useful methods to get the sensitivity value or the function reference (see the other unit tests)?

Reply:
Done.